trolldb/config/config.py

```diff
@@ -108,37 +109,38 @@ class AppConfig(BaseModel):
     subscriber: SubscriberConfig
 
 
-@validate_call
-def parse_config_yaml_file(filename: FilePath) -> AppConfig:
-    """Parses and validates the configurations from a YAML file.
+@logger.catch(onerror=lambda _: sys.exit(1))
+def parse_config(file: int | str | bytes | PathLike[str] | PathLike[bytes]) -> AppConfig:
+    """Parses and validates the configurations from a YAML file (descriptor).
 
     Args:
-        filename:
-            The filename of a valid YAML file which holds the configurations.
+        file:
+            A path-like object (``str`` or ``bytes``) or an integer file descriptor. This will be directly passed to the
+            ``open()`` function. For example, it can be the filename (absolute or relative) of a valid YAML file which
+            holds the configurations.
 
     Returns:
         An instance of :class:`AppConfig`.
-
-    Raises:
-        ParserError:
-            If the file cannot be properly parsed.
-
-        ValidationError:
-            If the ``filename`` is not of type ``FilePath``.
-
-        ValidationError:
-            If the successfully parsed file fails the validation, i.e. its schema or the content does not conform to
-            :class:`AppConfig`.
     """
-    logger.info("Attempt to parse the YAML file ...")
-    with open(filename, "r") as file:
-        config = safe_load(file)
+    try:
+        logger.info("Attempt to parse the YAML file ...")
+        with open(file, "r") as f:
+            config = safe_load(f)
+    except parser.ParserError as e:
+        logger.error(f"The file could not be parsed: {e}")
+        sys.exit(errno.EIO)
+    except (OSError, FileNotFoundError) as e:
+        logger.error(f"The file (descriptor) could not be found or opened: {e}")
+        sys.exit(errno.EIO)
```

Comment:
Is this really needed? if yaml can't parse the file, there will be an error anyway, with the added value that it will show where the parsing went wrong.

Same for the second error, python will raise it anyway.

"No code is better than no code"

Reply:
Done. The exception catching has been removed.